File: week02/day_2/draw_diamond.py
__author__ = 'ylwoi'

# Write a program that reads a number from the standard input, then draws a
# diamond like this:
#
#
#    *
#   ***
#  *****
# *******
#  *****
#   ***
#    *
#
# The diamond should have as many lines as the number was

x = int(input("Enter a number to set the diamond's size: "))
for i in range(x):
    print((x-i) * " " + (i * "*") * 2 + "*")

for j in range(x, -1, -1):
    print((x -j) * " " + (j * "*") * 2 + "*")

File: week07/day2/draw-diamond.js
/**
 * Created by cw on 2017-05-02.
 */
'use strict';

var lineCount = 7;
var star = '*';
var space = ' ';


// Write a program that draws a
// diamond like this:
//
//
//    *
//   ***
//  *****
// *******
//  *****
//   ***
//    *
//
// The diamond should have as many lines as lineCount is

for (var i = 0, j = 1, k = lineCount; i < lineCount; i++, j += 2, k--) {
    if (i < (lineCount-1)/2) {
        console.log(space.repeat(k), star.repeat(j));
    } else {
        for (;i < lineCount; i++, j -= 2, k++) {
            console.log(space.repeat(k), star.repeat(j));
        }
    }
}